python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
         """
         self._log = logger
 
-        (sg_app_store, script_user) = shotgun.create_sg_app_store_connection()
-        self._sg = sg_app_store
-        self._sg_script_user = script_user
+        from tank_vendor.shotgun_deploy import Descriptor, create_descriptor
 
-        self._local_sg = shotgun.get_sg_connection()
-        self._update_version = self.__get_core_version(core_version)
+        self._current_core_descriptor = create_descriptor(
+            shotgun.get_sg_connection(),
+            Descriptor.CORE,
+            {
+                "type": "app_store",
+                "version": pipelineconfig_utils.get_currently_running_api_version(),
```

Comment:
When we scheduled this ticket we had decided to only support app store descriptors for now since it was the immediate need. Can this wait until we revisit the command to be able to upgrade the core from any source?